presto-hive/src/main/java/com/facebook/presto/hive/HiveRecordSink.java

```diff
@@ -233,7 +240,9 @@ private void append(Object value)
         Type type = columnTypes.get(field);
         if (isMapType(type) || isArrayType(type)) {
             // Hive expects a List<>/Map<> to write, so decode the value
-            value = TypeJsonUtils.stackRepresentationToObject(connectorSession, (String) value, type);
+            Slice slice = Slices.utf8Slice((String) value);
```

Comment:
I think you need to special case MAP and ARRAY in appendString() otherwise it will break when there are non-utf8 characters in the encoded value. Then you can change this to be a cast to byte[] instead of String